examples/large_models/segment_anything_fast/custom_handler.py

```diff
@@ -0,0 +1,91 @@
+import base64
+import io
+import json
+import logging
+import pickle
+import zlib
+
+import cv2
+import numpy as np
+import torch
+from PIL import Image
+from segment_anything_fast import SamAutomaticMaskGenerator, sam_model_registry
+from segment_anything_fast.tools import apply_eval_dtype_predictor
+
+from ts.torch_handler.base_handler import BaseHandler
+
+logger = logging.getLogger(__name__)
+
+
+class SegmentAnythingFastHandler(BaseHandler):
+    def __init__(self):
+        super(SegmentAnythingFastHandler, self).__init__()
+        self.mask_generator = None
+        self.initialized = False
+
+    def initialize(self, ctx):
+        properties = ctx.system_properties
+        if torch.cuda.is_available() and properties.get("gpu_id") is not None:
+            self.map_location = "cuda"
+            self.device = torch.device(
+                self.map_location + ":" + str(properties.get("gpu_id"))
+            )
+
+        model_type = ctx.model_yaml_config["handler"]["model_type"]
+        sam_checkpoint = ctx.model_yaml_config["handler"]["sam_checkpoint"]
+
+        self.model = sam_model_registry[model_type](checkpoint=sam_checkpoint)
+        self.model.to(self.device)
+
+        self.mask_generator = SamAutomaticMaskGenerator(self.model)
+        self.mask_generator.predictor = apply_eval_dtype_predictor(
+            self.mask_generator.predictor, torch.bfloat16
+        )
+
+        logger.info(f"Model weights {sam_checkpoint} loaded successfully")
+        self.initialized = True
+
+    def preprocess(self, data):
+        images = []
+        for row in data:
+            image = row.get("data") or row.get("body")
+            if isinstance(image, str):
+                # if the image is a string of bytesarray.
+                image = base64.b64decode(image)
+
+            # If the image is sent as bytesarray
+            if isinstance(image, (bytearray, bytes)):
+                image = Image.open(io.BytesIO(image))
+            else:
+                # if the image is a list
+                image = torch.FloatTensor(image)
+
+            image = cv2.cvtColor(np.array(image), cv2.COLOR_RGB2BGR)
+            images.append(image)
+
+        return images
```

Comment:
Does dynamic shape work for this model or will we retrigger compilation for different sizes?

Reply:
SAM internally is resizing to 1024x1024. Tried with different sizes. Works without recompiling